aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -226,7 +225,13 @@ def _generate_hash(self, data: Any) -> str:
             Hashed representation of the provided data
 
         """
-        data = getattr(data, "raw_event", data)  # could be a data class depending on decorator order
+        if hasattr(data, "__dataclass_fields__"):
```

Comment:
@heitorlessa - not sure if there is a cleaner way of doing this without breaking python 3.6

Reply:
Luckily 3.6 is EOL in 6 days ;-) so we will be able to drop it fully soon (need to check with the Lambda team).

In the meantime, can we move this serialisation into a shared function -or private method for now- as we will later support Pydantic models too?

We will be reuse this serialisation for Logger and possibly others too, e.g. be able to serialize Event Source 
Data Class, Dataclasses, and Pydantic Model consistently within Powertools

Reply:
@heitorlessa - just as a dict or list? coz json serialization is not always consistent throughout

Reply:
Not sure I follow? You mean how are we going to handle the return if it's a list?

Reply:
Thanks for clarifying on messaging, let's keep as a private method - we want the dict only. 

Later, we can turn into a reusable function to get the dict value then serialize to JSON str for the encoder

Reply:
@heitorlessa - i have added  prepare method.